lib/xero_gateway/base_record.rb

```diff
@@ -0,0 +1,97 @@
+module XeroGateway
```

Comment:
We should consider using `ActiveModel` for this at some point rather than re-creating so much of it...